benchmarks/benchmark_distance_transform.py

```diff
@@ -0,0 +1,76 @@
+# See "Writing benchmarks" in the asv docs for more information.
+# https://asv.readthedocs.io/en/latest/writing_benchmarks.html
+import numpy as np
+from scipy import ndimage as ndi
+from scipy.ndimage.morphology import distance_transform_edt
+from skimage.transform import generalized_distance_transform
+
+
+class edt2d:
+    """Benchmark for distance transform in scikit-image."""
+    timeout = 120.0
+    def setup(self):
+        self.case = (1+-1*(np.random.randint(50, size=(4096,4096))//49)).astype('float64')
```

Comment:
Is it possible for you to reduce the size of these arrays? I think you should still see your improvement.

Reply:
right, so here comes the challenging part. 

I think for fairness you should keep both: 
1. Small
2. Large

See this for the syntax
https://github.com/numpy/numpy/blob/master/benchmarks/benchmarks/bench_shape_base.py#L10

If scipy is 30x faster, it might be worthwhile to have a `scipy-fallback` for small images. both use cases are quite popular, 256x256 (small) images, and 1024 x 1024 (large) images.

Reply:
I think you can use that syntax to actually avoid duplicating your 2d/3d code.

Reply:
I think we have a big overhead compared to scipy *per 1D scan*, so you only see the benefits of this new algorithm when your scanlines are big enough. I guess the way to solve this would be to Cythonise "apply_along_axis". Or, rather, to profile the code on a small array and figure out what we need to speed up.